src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -563,7 +564,9 @@ private BlittableJsonReaderObject AddCounterAndTimeSeriesSnapshotsIfNeeded(Docum
                     return null;
                 }
 
-                document = ChangeSnapshotFlag(context, document, id, Constants.Documents.Metadata.RevisionCounters, Constants.Documents.Metadata.Counters);
+                (document, bool documentChanged) = ChangeSnapshotFlag(context, document, id, Constants.Documents.Metadata.RevisionCounters, Constants.Documents.Metadata.Counters);
+                if (documentChanged)
+                    flags = oldFlags.Strip(DocumentFlags.Revision | DocumentFlags.DeleteRevision | DocumentFlags.HasTimeSeries);
```

Comment:
why it was not needed before the change of 4.2?

Reply:
because we haven't changed the document

Reply:
But we only remove one flag fun the metadata, no?
How is that different from what we had in 5.2 previously?

Reply:
Done - you are right, we only need to handle `DocumentFlags.HasCounters`